### PR TITLE
Add Github Actions CI workflow.

### PR DESCRIPTION
**Pull request by teymour-aldridge**
From what I understand, TravisCI no longer offer CI for open source projects without jumping through a lot of hoops to obtain free credits.

### CODE REVIEW

**Comment by cfvescovo** [comment]
Looks good to me

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        rust_version: [1.36.0, stable, beta, nightly]
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install latest nightly
+        uses: actions-rs/toolchain@v1
+        with:
+            toolchain: ${{matrix.rust_version}}
+            override: true
+      - uses: Swatinem/rust-cache@v1
```
Comment (cfvescovo) [inline .github/workflows/test.yml:39]: Maybe we should use the actions/cache@v2 action.
Consider this example:
```
- uses: actions/cache@v2
  with:
    path: |
      ~/.cargo/bin/
      ~/.cargo/registry/index/
      ~/.cargo/registry/cache/
      ~/.cargo/git/db/
      target/
    key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}

```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install latest nightly
+        uses: actions-rs/toolchain@v1
+        with:
+            toolchain: stable
+            override: true
+            components: rustfmt, clippy
+      - uses: Swatinem/rust-cache@v1
```
Comment (cfvescovo) [inline .github/workflows/test.yml:21]: As pointed out below, consider using `actions/cache@v2` instead of `Swatinem/rust-cache@v1`

**Comment by cfvescovo** [comment]
@teymour-aldridge Moreover, I think we should not test on 1.36 anymore

**Comment by teymour-aldridge** [comment]
That seems true. I'll fix that (unless you'd like to submit a patch for it).

**Comment by cfvescovo** [comment]
Thank you. I am working on updating the dependencies and switching the Rust language edition to 2021

**Comment by teymour-aldridge** [comment]
Great!

**Comment by cfvescovo** [comment]
Let me know when your code is ready, I will check it out asap.

**Comment by teymour-aldridge** [comment]
Oh sorry, I misunderstood; I thought you were working on a fix lol :)

I'll submit a patch as soon as I am able (fair warning, I'm a bit beleaguered with work at the moment).

**Comment by cfvescovo** [comment]
I have time to fix this now. I will open a PR soon

**Pull request merged**